src/internal/operations/backup.go

```diff
@@ -516,6 +516,38 @@ func matchesReason(reasons []kopia.Reason, p path.Path) bool {
 	return false
 }
 
+// getNewPathRefs returns
+//  1. the new RepoRef for the item if it needs merged
+//  2. the new locationPath
+//  3. if the location was likely updated
+//  4. any errors encountered
+func getNewPathRefs(
+	dataFromBackup kopia.DetailsMergeInfoer,
+	entry *details.DetailsEntry,
+	repoRefPB *path.Builder,
+	backupVersion int,
+) (path.Path, *path.Builder, bool, error) {
+	// Right now we can't guarantee that we have an old location in the
+	// previous details entry so first try a lookup without a location to see
+	// if it matches so we don't need to try parsing from the old entry.
+	//
+	// TODO(ashmrtn): In the future we can remove this first check as we'll be
+	// able to assume we always have the location in the previous entry. We'll end
+	// up doing some extra parsing, but it will simplify this code.
+	newPath, _, newPrefix := dataFromBackup.GetNewPathRefs(repoRefPB, nil)
```

Comment:
Nit: we already use the term `prefix` to primarily speak about the repo-ref prefix.  This helps separate the vocabulary, just a little.
```suggestion
	newPath, _, newLocPrefix := dataFromBackup.GetNewPathRefs(repoRefPB, nil)
```